### PR TITLE
QA code improvements to discussion.js, admin.js, lesson.js and workshop.js.

### PR DESCRIPTION
**Pull request by carmen12**

### CODE REVIEW

File: app/lib/audit.js [hunk 1]
```diff
@@ -153,9 +162,24 @@ function validate_record(r, k, ln){
...
 var logs = {},
-    file = fs.createWriteStream('/opt/data/audit_log.tsv');
+    file = fs.createWriteStream('/opt/data/audit_log.tsv'),
+    file_sql = fs.createWriteStream('/opt/data/audit_log.sql');
+
+file_sql.write("INSERT INTO mdl_log (time,userid,ip,course,module,cmid,action,url,info) VALUES ");//the first part of the INSERT command will be written only once, to make execution faster
 
 function audit(table, module, action){
+    fs.readFile('/opt/data/audit_log.sql', function (err,data) {//needed to add a ";"" at the end of the sql commands
+      if (err) { return console.log(err);}
```
Comment (jlabusch) [inline app/lib/audit.js]: Indenting

File: app/lib/audit.js [hunk 2]
```diff
@@ -153,9 +162,24 @@ function validate_record(r, k, ln){
...
 
 var logs = {},
-    file = fs.createWriteStream('/opt/data/audit_log.tsv');
+    file = fs.createWriteStream('/opt/data/audit_log.tsv'),
+    file_sql = fs.createWriteStream('/opt/data/audit_log.sql');
+
+file_sql.write("INSERT INTO mdl_log (time,userid,ip,course,module,cmid,action,url,info) VALUES ");//the first part of the INSERT command will be written only once, to make execution faster
 
 function audit(table, module, action){
+    fs.readFile('/opt/data/audit_log.sql', function (err,data) {//needed to add a ";"" at the end of the sql commands
```
Comment (jlabusch) [inline app/lib/audit.js]: This seems wrong... Surely we shouldn't read+write a file in passing every time we access append() or flush()?

**Review by jlabusch — CHANGES_REQUESTED** [review]
Let's talk, maybe I'm misunderstanding something.

**Comment by carmen12** [comment]
I've added the new solution that we decided on this morning. Seems to be working well now.

**Pull request merged**